modules/AOW_WorkFlow/AOW_WorkFlow.php

```diff
@@ -755,8 +788,28 @@ public function check_valid_bean(SugarBean $bean)
         return true;
     }
 
+    /**
+     * @param SugarBean $bean
+     * @return bool
+     */
+    protected function hasAlreadyRunToday(SugarBean $bean)
+    {
+        global $db, $timedate;
+
+        $day = $timedate->getNow()->format('Y-m-d');
+        $sql = "SELECT id FROM aow_processed WHERE aow_workflow_id = '{$this->id}' " .
+            "AND parent_id = '{$bean->id}' AND status = 'Complete' AND date_entered LIKE '$day%' LIMIT 1";
+
+        if ($db->fetchByAssoc($db->query($sql))) {
+            return true;
+        }
+        return false;
+    }
+
     public function compare_condition($var1, $var2, $operator = 'Equal_To')
     {
+        global $timedate;
```

Comment:
do not use superglobals or globals if it's not necessary, pass it as an (optional) dependency.